### ISSUE TITLE
wrong number of arguments (2 for 1) (ArgumentError) 

### ISSUE THREAD

**Issue by ghost** [sporkrb/spork#174]
When starting Spork with diagnose option as in:

`bundle exec spork -d`

I get the following error message:

```

Using RSpec, Unknown
Loading Spork.prefork block...
wrong number of arguments (2 for 1) (ArgumentError)
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/diagnoser.rb:35:in `load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/activesupport-3.2.1/lib/active_support/dependencies.rb:245:in `block in load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/activesupport-3.2.1/lib/active_support/dependencies.rb:236:in `load_dependency'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/activesupport-3.2.1/lib/active_support/dependencies.rb:245:in `load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/engine.rb:588:in `block (2 levels) in <class:Engine>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/engine.rb:587:in `each'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/engine.rb:587:in `block in <class:Engine>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/initializable.rb:30:in `instance_exec'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/initializable.rb:30:in `run'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/initializable.rb:55:in `block in run_initializers'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/initializable.rb:54:in `each'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/initializable.rb:54:in `run_initializers'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/application.rb:136:in `initialize!'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/railties-3.2.1/lib/rails/railtie/configurable.rb:30:in `method_missing'
/Users/potiguar/src/delta/config/environment.rb:5:in `<top (required)>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/diagnoser.rb:31:in `require'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/diagnoser.rb:31:in `require'
/Users/potiguar/src/delta/spec/spec_helper.rb:6:in `block in <top (required)>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork.rb:23:in `prefork'
/Users/potiguar/src/delta/spec/spec_helper.rb:4:in `<top (required)>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/diagnoser.rb:37:in `load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/diagnoser.rb:37:in `load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/test_framework.rb:138:in `block (2 levels) in preload'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/app_framework.rb:49:in `preload'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/test_framework.rb:134:in `block in preload'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork.rb:61:in `exec_prefork'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/test_framework.rb:120:in `preload'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/runner.rb:69:in `run'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/lib/spork/runner.rb:10:in `run'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/gems/spork-1.0.0rc1/bin/spork:10:in `<top (required)>'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/bin/spork:19:in `load'
/Users/potiguar/.rvm/gems/ruby-1.9.3-p0/bin/spork:19:in `<main>'
```

**Comment by andreacfm** [comment]
Same for me!

**Comment by kasima** [comment]
Same here!

On ruby 1.9.2-p290, Rails 3.2

**Comment by cola-zero** [comment]
Same on Spork 0.9.0, Rails 3.2.1, 1.9.3-p0, and @fromagie 's patch (bf4348c) is work for me.

**Comment by SkyWriter** [comment]
@fromagie thanks a lot. Worked for me either.

**Comment by relaxdiego** [comment]
@fromagie's patch fixed it for me as well. thanks!

**Comment by xxx** [comment]
+1 same issue on 0.9.0 and 1.0.0rc2. patch fixes.

**Comment by nickrivadeneira** [comment]
@fromagie Patch fixed for me, too.

**Comment by flyerhzm** [comment]
+1

**Comment by mcmire** [comment]
I can confirm @fromagie's patch fixes this too. Patch applies cleanly on master. @timcharper can you consider merging this?

**Comment by hube** [comment]
@fromagie's patch fixed this issue for me on 1.0.0rc2

**Comment by stefanwille** [comment]
+1

**Comment by timcharper** [comment]
Merged fix and deployed new versions of 0.9.0 and 1.0.0.

**Comment by mcmire** [comment]
Awesome, thanks so much.
